backend/providers.py

```
@@ -0,0 +1,584 @@
+        except Exception as e:
+            logger.error("Failed to list Ollama models: %s", e)
+            return []
+
+    # -- health_check -------------------------------------------------------
+
+    def health_check(self) -> Dict[str, Any]:
+        try:
+            resp = requests.get(f"{self.base_url}/api/tags", timeout=5)
+            resp.raise_for_status()
+            model_count = len(resp.json().get("models", []))
+            return {"status": "ok", "provider": "ollama", "models_available": model_count}
```

Ruff (0.15.10)

[warning] Consider moving this statement to an `else` block

(TRY300)